lib/kubernetes-deploy/render_task.rb

```diff
@@ -29,7 +29,7 @@ def run!(stream, only_filenames = [])
       @logger.phase_heading("Initializing render task")
 
       filenames = if only_filenames.empty?
-        TemplateDiscovery.new(@template_dir).templates
+        TemplateDiscovery.new([@template_dir]).templates[@template_dir]
```

Comment:
This is the result of the change in contract of `TemplateDiscovery#new` requiring an array of template dirs. 
- Do we want the render task to accept multiple template dirs. Note that the render task can also accept an arbitrary list of filenames as a filter. For multiple template dirs, we'll just apply the filter to each directory.
- If we do want the render task to accept multiple template directories as well, do we want to tackle that in this PR or in a follow-up?

Reply:
I think you're absolutely right. The current semantics of `kubernetes-render` are incompatible with a `kubectl -f`-like model. For instance, there would be no way to know which bare file names provided to `kubernetes-render` point to which provided template dirs :/

To me, this means we need to make a fundamental change to the CLI contract of `kubernetes-render`. In particular, dropping support for providing bare filenames as arguments and instead force the (conventional) semantics of the `-f` flag. From your above example, this would be:
`krane render -f dirA/one.yml,dirB/two.yml` (not that bad).

In light of this, I'd like to keep `kubernetes-render` _as-is_ (only supporting a single template dir), and make a follow-up PR to make `krane render` behave with `-f` semantics.